src/model/mod.rs

```diff
@@ -1 +1,290 @@
+use term_painter::ToStyle;
+use term_painter::Color::*;
+use std::io;
+
+mod types;
+
 // Game logic (board, initialization, valid move, set, play or finished)
+
+/// Visualization of the boards.
+fn print_boards(board1: &Vec<types::SubField>, board2: &Vec<types::SubField>) {
+    println!("{}{}", "------------------ O W N   B O A R D ------------------",
+        "-------------------------- O P P O N E N T ---------------");
+
+    let mut cnt = 9;
+
+    for i in 0..10 {
+        println!("");
+        print!("{}  ", cnt);
+        for j in 0..10 {
+            if board1[j + i * 10] == types::SubField::Hit {
+                print!(" {} ", Red.paint(board1[j + i * 10]));
+            } else {
+                print!(" {} ", board1[j + i * 10]);
+            }
+        }
+        print!("        ");
+        print!("{}  ", cnt);
+        for j in 0..10 {
+            if board2[j + i * 10] == types::SubField::Hit {
+                print!(" {} ", Green.paint(board2[j + i * 10]));
+            } else {
+                print!(" {} ", board2[j + i * 10]);
+            }
+        }
+        println!("");
+        cnt -= 1;
+    }
+    println!("");
+    println!("{}{}", "     A    B    C    D    E    F    G    H    I    J",
+        "               A    B    C    D    E    F    G    H    I    J");
+    println!("");
+    println!("{}{}", "----------------------------------------------------",
+        "-------------------------------------------------------------");
+}
+
+/// Reads a string from the terminal / user.
+fn read_string() -> String {
```

Comment:
I'd actually go for a "terminal_io.rs". We could also move the print_boards function there as both methods are terminal-only and we could just "get rid" of them once we have the gui